runtime/caml/sizeclasses.h

```diff
@@ -4,6 +4,9 @@
 #define SIZECLASS_MAX 128
 #define NUM_SIZECLASSES 32
 
+typedef unsigned char sizeclass_t;
+static_assert(NUM_SIZECLASSES < (1 << (CHAR_BIT * sizeof(sizeclass_t))), "");
```

Comment:
`*_t` type names are reserved by POSIX, maybe we should avoid introducing a new one. Could the `static_assert` or the code be simplified by switching to a `uint8_t`?

Reply:
I mostly ignore the POSIX reservation (because (a) the set of actual POSIX typenames is pretty well-known, and (b) `_t` is fairly widespread in the wild, and (c) I like `_t` as a comprehension aid - the reservation annoyed me at the time and annoys me still), but maybe we should have a policy.
Re `uint8_t`: I'm not opposed to that at all; I was just sticking with the existing type. Either way the `static_assert` could be simplified (because `sizeof(char)` is defined to be 1, and `uint8_t` is defined to have 8 bits); I wrote it that way so it stands on its own, independently of the `typedef`, but maybe that's just too conservative.

Reply:
I've changed my `sizeclass_t` to `sizeclass`.

Reply:
If POSIX says `*_t` is reserved we should abide, even if it's annoying.

As for `asize_t`, it was so long ago that I don't even remember who introduced it and what the `a` stands for (maybe "address").